### DISCUSSION TITLE
If you got locked out of the refer-a-friend feature and the 200% xp bonus or have an issue accessing it please report your experience here

### DISCUSSION THREAD

**Discussion by rsepierre** [chromiecraft/chromiecraft#2826] [General]
### This is an subject for the community :

Today, gameplay-wise there are two types of players accounts :
- Players with 200% boost
- Players with 100% rate

This could make in-game time difference to reach the same objectives of DAYS of even WEEKS (considering multiple characters per accounts). Which I think makes is a defining/game breaking feature and very important to discuss. As a RAF account is simply better than a non-RAF account.

**Please share your experience if you have any issue accessing the refer-a-friend feature. Admins allows you to ask for codes freely in chat even if no one refered you so it is basicaly open for anyone**
- Note that if you refer yourself, you might get the 200% rested xp for a few days before it getting revoked and permanently locked out of your account and loose all progress on your account (as in : it will probably be faster to level up creating a newer account).
- You should always then properly exploit the feature (use different IP) which is tolerated by admins as long as you don't get caught by the automatic IP monitoring.
- Or more conveniently you should ask in discord or world chat for a random referal code because it doesn't matter if you really got refered or not.
- If your friend is your roomate, your boy/girlfriend or a family member, you should pretend that you don't know each other and ask for a code from a random person in chat giving him the reward.
- Also very important : If you refered an IRL friend and you were thinking about bringing your laptop to their house for a game sessions : please avoid it. Even after activation, your refer a friend status might be revoked if you share the same IP. It is very important for ChromieCraft's team that players enjoy the game alone.

I originaly reported the subject in the bug-tracker as you can use software methods to exploit the feature and game in-game advantage. But admins says it's not a bug, not an issue, not an exploit.

__Link my original post on bug tracker which discussed what I think are the main issues with the current implementation of RAF:__
https://github.com/chromiecraft/chromiecraft/issues/2823

__Link to my anwser to @daliss with a step-by-step guide to easely benefit from the 200% xp boost without a referal code :__
https://github.com/chromiecraft/chromiecraft/discussions/2818#discussioncomment-1991129

Other questions to the community : 

Do you think the refer-a-friend feature is nicely implemented ?
Is the fact that it is easely exploited a problem ?
Should there be alternative ways to get the same 200% xp as RAF users/exploiters ?
Should people who discover the server on their own grind twice as mush as people who got invited ?
Should people ask for referal codes in /world chat from non-friends ?
Should it be renamed to the "ask for 200% xp in /world" feature ?
Should you be able to refer your brother next room ?
Should you be able to self-refer ?
Should you be permanently locked out of the refer-a-friend feature if you tried to self-refer or still be able to ask someone new to refer you ?
Would any one be happy to play with 100% xp rate (and not have the choice) when most players are at 200% ?

**Comment by rsepierre** [comment]
The reason I'm persistent is because no admin actually read the post before throwing it off. Yes it works fine for 99% of people. It's not because it's bad for 1% that it's not an issue. Accessibility and fairness should be built into the rules by design. And unecessary limitations which hinders even few player's experience  (that's still two in the last 24 hours) should be considered before being thrown out unread.

I don't know why @55Honey you locked Daliss discussion preventing people from engaging in the conversation. But there is no reason to. Only persons interested in the subject would take the time to contribute. 

I very strongly believe some aspects of the implementation should be improved in such a good project as azerothcore and ChromieCraft and I think it's natural to reach out to other player experiencing same or related issues as you to discuss it further and make our views heard.

**Comment by rsepierre** [comment]
![B17DACA9-F48A-42DF-9E8F-0FB575594B8E](https://user-images.githubusercontent.com/6746875/150009079-bb034e69-2051-4f01-ba74-5a7152a4d29f.jpeg)
![7124A808-3DE7-4C65-A78B-D3E88DF41B6D](https://user-images.githubusercontent.com/6746875/150009086-97ecd1fa-5353-429e-8643-f0efe9c8d0bb.jpeg)
![FA148BE6-BB9C-4DA6-8526-7D9E7BF33AE7](https://user-images.githubusercontent.com/6746875/150009090-d3a703e8-9104-427d-9d4a-528f026accd2.jpeg)

**Comment by rsepierre** [comment]
![F12F2166-2C58-4C99-BC0A-B31C36762528](https://user-images.githubusercontent.com/6746875/150010086-4ac9088e-2737-493e-8c75-625947e53506.jpeg)

**Reply by rsepierre** [reply]
![55BDEB10-F501-43F6-98B8-DBC44C764B03](https://user-images.githubusercontent.com/6746875/150010320-bdf6b568-0db1-4cee-827f-908d3e5b1dd6.jpeg)

**Comment by rsepierre** [comment]
![C47183E6-1F88-42C8-8E1A-20707A25FB95](https://user-images.githubusercontent.com/6746875/150010484-71f7a842-12ab-4961-b139-c65482bcf778.jpeg)

**Comment by rsepierre** [comment]
imagine having some accounts with a net advantage over other accounts

![0B89BFC8-F5D7-4CCA-AC22-CEA277537A5F](https://user-images.githubusercontent.com/6746875/150010885-26b1a76f-fe6b-46c2-97de-1e8be1e69039.jpeg)

**Comment by rsepierre** [comment]
![A95FBF68-AC21-4616-8D7A-4C9001AD1993](https://user-images.githubusercontent.com/6746875/150011765-d48ed5b9-a560-4b66-898e-947134f84258.jpeg)
![06B884BA-89E2-4029-B810-19372E56D640](https://user-images.githubusercontent.com/6746875/150011775-8beed91d-5f50-42f5-bfd0-7a313c0dd0e5.jpeg)

**Reply by Amrx2** [reply]
Pierre, you can continue in this vain but it will not change the RAF bonus which has worked as we have wanted it for months. You have been given a way for you to receive this benefit and we aren't changing our exp rates either. Thank you, Construct.

**Reply by rsepierre** [reply]
The way to recieve the benefit need for me to delete my current progress and it happens to other players. I understand that the raf feature works perfectly for you and it's aim and I'm not trying to make you consider changes to make it to any state that is "worse" and but only those for the better. 
I also understand that my voice alone will not be sufficient. This is why I'm regrouping past conversations over here and invite future people taking issue with the feature to take it here.

**Reply by Amrx2** [reply]
And again, you are welcome to but it has been created by Honey55 to the specs we require and isn't changing. The module is open source like everything we do so feel free to check it out.

**Reply by rsepierre** [reply]
As we didn't hear from Daliss since he has been answered negatively in his own topic on the same matter, I fear he left the server for ever in disatisfaction. Just as I would have if I wasn't stupid and stubborn to try to find solutions for devs and players.

**Reply by rsepierre** [reply]
As I finished regrouping related community reactions I will leave both the server and discussions for a while only reacting to new developments. I simply encourage people experiencing issues such as being unable to refer a family member, being blocked-out or successfully self-refering without getting into trouvle to post about it here.

Rested XP is not a small in-game advantage and shouldn't be treated lightly. And locking some players out of the feature can badly affects experience. Every feedback bit of feedback important. Hopefully me and Daliss are the only one ever having issues with the feature.

unlucky for him, his thread was closed and now inaccessible for further comments.

**Reply by Amrx2** [reply]
Again, and for the last time, whilst the player can use RAF we have it setup and made by the team here (yes the entire module) for a reason and the player needs to meet specific criteria to use it. I've already told you before in world chat some of the reasoning behind it and won't bother to repeat it as you never take it in nor look at it from any other angle other than yourself (I spent some time checking the world chat logs and you had 2 people support you and over 50 against). 

Yes feedback is important but you have got to understand not all feedback is implemented nor useful to the server in the bigger picture of things. 

I'll put it in simple terms, we are a seafood restaurant. You are demanding a pork chop. We explain we can offer you a surf & turf only. You then bring up prior examples of a small handful of people unhappy we couldn't offer you a pork chop, ignoring the thousands who had enjoyed that surf & turf. In life it is impossible to please everyone whilst you do your own thing, pleasing 99% however whilst still getting the analytics benefit out of RAF and the need to use codes behind it is still a win for thousands. 

So again, thank you for the feedback but RAF is remaining as it is and for good reason. Cheers.

**Reply by rsepierre** [reply]
"the player needs to meet specific criteria to use it." The only criteria not needed to be met is actually to refer a friend (as some ranked people have referal codes in their names and you can get codes in world chat).

You never developed the "good reasons" part and how it would not be possibly improved. That's maybe why I don't understand them. Why you monitor users IP ? And how the fact that Saliss tried to self-refer instead of asking world chat thus getting permanntly locked out of the feature is better for analytics and user experience.

**Reply by rsepierre** [reply]
Why is there a policeman in your seafood restaurant (monitoring ips) if the owner is pretty chill about you bringing your own pork to the table

**Reply by rsepierre** [reply]
If you stop banning people out of the feature ( just remove link for example but being able to ask in chat for a new one ) the feature doesn't "change" drastically. It still serves the same purpose. Nobody get's frustrated. Nobody loses progress. It will still perform as good in analytics and will require less ressource to run.

**Reply by Amrx2** [reply]
Again, this is our house and we have set it up as we want it to be to bring the best experience we can, creating the best core and offering it for free for all. We are a proud seafood restaurant and it's only surf and turf, no pork thanks. The RAF was an added benefit later on so people new to Chromiecraft can use it to level up to others faster or to create new toons and level them faster on a new account. 

I never mentioned anything about monitoring IPs either. Yet again, we see a lot of analytics who brings in people, where are they coming from, are they using RAF or not. We can then use that info to post about Chromiecraft and our open source philosophy in that community. Hopefully that way we get more players on board who then might turn into contributors helping us on our goal as stated earlier (https://www.chromiecraft.com/how-to-contribute/)  The majority keep on x1 rates actually and don't use the free rested btw. That is just one benefit of many we get from RAF being in this method. RAF is fully automated as well, like I said the module is here so feel free to look it up.

If you don't like the fact we won't serve you pork, you can try out any of those 20 servers you mentioned about in world to get that feeling of a good pork chop Pierre. 

For the very final time, RAF is here to stay as it is. When we get to TBC, yes we might look at boosting rates in vanilla as an option. Until then, maybe another exp event in the future but we just had one over Christmas so will be a while again.

Thanks for your feedback and your concerns.

**Comment by rsepierre** [comment]
![E5420CC4-94AA-4F3C-A4B6-6D6D48FD432D](https://user-images.githubusercontent.com/6746875/150094155-61085fac-847b-4c68-a262-546e5778fc68.jpeg)

**Comment by 55Honey** [comment]
Since you're just spamming and not even reacting at all to replies, this is not a discussion. I think you've presented your conerns far beyond the point of exxaggeration. Due to you ignoring any reply, I can't even tell, if you're trolling at this point. 

I will close the topic now, since it does not contribute to the project in any good way.

**Comment by 55Honey** [comment]
Unfortunately, rsepierre has managed to spam so many channels and platforms and create so much drama about this subject, turning every single discussion back to this subject out of the blue, that we had no other choice than to block them from our GitHub.
